core/src/Kokkos_Half.hpp

```diff
@@ -1005,6 +1007,163 @@ cast_from_bhalf(bhalf_t val) {
 #else
 #define KOKKOS_BHALF_T_IS_FLOAT false
 #endif  // KOKKOS_IMPL_BHALF_TYPE_DEFINED
+////////////// BEGIN HALF_T (binary16) limits //////////////
+// bits [10,14] set
+// #define KOKKOS_IMPL_HALF_T_HUGE_VALH 0x7c00 0x7c00
+#if defined(KOKKOS_HALF_T_IS_FLOAT) && !KOKKOS_HALF_T_IS_FLOAT
+// Minimum normalized number
+template <>
+struct Kokkos::Experimental::Impl::finite_min_helper<
+    Kokkos::Experimental::half_t> {
+  static constexpr float value = -65504.0F;
```

Comment:
float -> half_t

Reply:
Is there a way to instantiate floating_point_wrapper at compile time?

Reply:
you'd need to make sure the constructor is constexpr

Reply:
@dalg24 Is https://github.com/kokkos/kokkos/pull/5778/commits/1b399bdd43da2b648d5f7cedbade0dfff51f8e43 what you had in mind?

Reply:
something like that yes